### PR TITLE
PCHR-3580: Fix bugs for multimple request per day

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

This PR fixes the following issues:

1) You should NOT be able to add a leave request in Hours and then add a leave request in days for the same day.
2) You should be able to create Hours TOIL Accrual after Days TOIL Accrual

### Issue 1 

When checking for existing AM/PM it should not care about requests in hours.

### Issue 2

TOIL should not care about neither requests types, not units. It should care only about dates/times of TOIL requests.

----

PHP tests augmented and passed

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php [hunk 1]
```diff
@@ -920,28 +920,31 @@ public static function findOverlappingLeaveRequests($leaveRequestParams, $leaveR
...
+        $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
       }
     }
 
-    if (!$isCalculationUnitInHours && !$isTOIL) {
-      $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
-    }
-
+    $ignoreRequestsWithDifferentUnit = count($absenceTypesIDsWithDifferentUnit)
+      ? 'OR lr.type_id IN (' . implode(',', $absenceTypesIDsWithDifferentUnit) . ')'
```
Comment (tunbola) [inline uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php]: Shouldn't it start with an 'AND'?

File: uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php [hunk 2]
```diff
@@ -920,28 +920,31 @@ public static function findOverlappingLeaveRequests($leaveRequestParams, $leaveR
...
+      : '';
     $considerHalfDayRequests =
       ($fromDateType === $halfDayPMID
         ? "AND (DATE_FORMAT(lr.to_date, '%Y-%m-%d') != '{$fromDate}' " .
-          "OR lr.to_date_type != {$halfDayAMID})"
+          "OR lr.to_date_type != {$halfDayAMID} {$ignoreRequestsWithDifferentUnit})"
```
Comment (tunbola) [inline uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php]: This query is not easy to understand, would have been easier to add the condition for NOT over lapping and negate it just like I suggested in the first PR review.

File: uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php [hunk 3]
```diff
@@ -1808,6 +1808,108 @@ public function testALeaveRequestShouldNotBeCreatedWhenThereAreOverlappingLeaveR
...
+    $contactId = 1;
+    $date = '2018-05-07';
+    $absenceTypeInDays = AbsenceTypeFabricator::fabricate();
+    $absenceTypeInHours = AbsenceTypeFabricator::fabricate(['calculation_unit' => 2]);
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id', 'validate'));
+    $halfDayPMId = $this->leaveRequestDayTypes['half_day_pm']['value'];
+
+    $period = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31')
```
Comment (tunbola) [inline uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php]: please remove extra spaces between the `=>`

File: uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php (hunk 1 above)
Comment (tunbola) [reply uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php]: Also is it actually ignoring requests with different units or taking them into consideration?

File: uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php (hunk 2 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php]: Attempted to improve.

File: uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php (hunk 1 above)
Comment (igorpavlov) [reply uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php]: Should be OR . Changed to "consider" - I think that's what was confising.

**Pull request merged**